### PR TITLE
V3: Transformer bindings

### PR DESCRIPTION
**Pull request by alshdavid**
- Added one-time setup to transformers
- Added `Transformer.loadConfig()`
- Expanded compat API
  - `AssetSymbols`
  - `MutableAssetSymbols` 
  - `PluginConfig`
  - `PluginOptions` (partial)
  - `PluginLogger` (basic)
  - `PluginTracer` (stub)
  - `MutableAsset`
- Updated `worker` transformer transform flow to use new compat structures
- Fixed most of the typing errors

### CODE REVIEW

File: crates/atlaspack_plugin_rpc/src/nodejs/plugins/nodejs_rpc_transformer.rs [hunk 1]
```diff
@@ -48,6 +53,9 @@ impl NodejsRpcTransformerPlugin {
       rpc_workers,
       plugin: plugin.clone(),
       plugin_options,
+      specifier: plugin.package_name.clone(),
+      resolve_from: (&*plugin.resolve_from).to_path_buf(),
+      project_root: (&*ctx.options.project_root).to_path_buf(),
```
Comment (alshdavid) [inline crates/atlaspack_plugin_rpc/src/nodejs/plugins/nodejs_rpc_transformer.rs:58]: Will clean this up in a following PR

File: packages/core/core/src/atlaspack-v3/worker/compat/plugin-options.js
Comment (alshdavid) [inline packages/core/core/src/atlaspack-v3/worker/compat/plugin-options.js:1]: Will remove getters when this object is more flashed out

File: crates/atlaspack_plugin_rpc/src/nodejs/plugins/nodejs_rpc_transformer.rs [hunk 2]
```diff
@@ -64,14 +72,23 @@ impl TransformerPlugin for NodejsRpcTransformerPlugin {
     _context: TransformContext,
     asset: Asset,
   ) -> Result<TransformResult, Error> {
+    self.rpc_workers.exec_on_all(|worker| {
+      worker.load_plugin(LoadPluginOptions {
+        kind: LoadPluginKind::Transformer,
+        specifier: self.specifier.clone(),
+        resolve_from: self.resolve_from.clone(),
+      })
+    })?;
```
Comment (alshdavid) [inline crates/atlaspack_plugin_rpc/src/nodejs/plugins/nodejs_rpc_transformer.rs:81]: Run this within a `OnceLock`

Comment (alshdavid) [reply crates/atlaspack_plugin_rpc/src/nodejs/plugins/nodejs_rpc_transformer.rs:86]: Done

File: crates/atlaspack_plugin_rpc/src/nodejs/plugins/nodejs_rpc_transformer.rs [hunk 3]
```diff
@@ -131,7 +153,7 @@ pub struct RpcPluginOptions {
 #[serde(rename_all = "camelCase")]
 pub struct RpcAssetResult {
   pub id: String,
-  pub bundle_behavior: BundleBehavior,
+  pub bundle_behavior: Option<BundleBehavior>,
```
Comment (mattcompiles) [inline crates/atlaspack_plugin_rpc/src/nodejs/plugins/nodejs_rpc_transformer.rs:156]: This should be `MaybeBundleBehaviour`

**Review by mattcompiles — APPROVED** [review]

File: crates/atlaspack_plugin_rpc/src/nodejs/plugins/nodejs_rpc_resolver.rs [hunk 4]
```diff
@@ -56,20 +57,19 @@ impl ResolverPlugin for RpcNodejsResolverPlugin {
...
       self.rpc_workers.exec_on_all(|worker| {
         worker.load_plugin(LoadPluginOptions {
           kind: LoadPluginKind::Resolver,
           specifier: self.specifier.clone(),
           resolve_from: self.resolve_from.clone(),
         })
-      })?;
-    }
+      })
+    })?;
```
Comment (alshdavid) [inline crates/atlaspack_plugin_rpc/src/nodejs/plugins/nodejs_rpc_resolver.rs:72]: This will only run the first time `Transformer.transform` is called

**Pull request merged**